### PR TITLE
Reset zeroData on new data requests.

### PR DESCRIPTION
**Pull request by tofumatt**
## Summary

Addresses issue #184.

## Checklist

- [x] My code is tested and passes existing unit tests.
- [x] My code has an appropriate set of unit tests which all pass.
- [x] My code is backward-compatible with WordPress 4.7 and PHP 5.6.
- [x] My code follows the [WordPress](https://make.wordpress.org/core/handbook/best-practices/coding-standards/) coding standards.
- [x] My code has proper inline documentation.
- [x] I have signed the Contributor License Agreement (see <https://cla.developers.google.com/>).

### CODE REVIEW

File: assets/js/components/higherorder/withdata.js [hunk 1]
```diff
@@ -204,6 +204,7 @@ const withData = (
 					data: returnedData,
 					datapoint,
 					module: identifier,
+					zeroData: !! isDataZero( returnedData, datapoint, requestData ),
```
[the remark this answers is not in the document]
Comment (tofumatt) [reply assets/js/components/higherorder/withdata.js:207]: Ah, fair. Sorry, the code paths for those modules is sometimes I bit tough for me to follow, but that makes sense. Cheers! 👍

**Comment by tofumatt** [comment]
All set for another review; made the change and it still worked a treat, so let's do the reset in the hook as mentioned 😄

File: assets/js/components/higherorder/withdata.js [hunk 2]
```diff
@@ -151,7 +151,7 @@ const withData = (
 				'googlesitekit.moduleDataReset',
 				'googlesitekit.moduleDataResetHandler',
 				() => {
-					this.setState( { data: false } );
+					this.setState( { data: false, zeroData: false } );
```
Comment (aaemnnosttv) [inline assets/js/components/higherorder/withdata.js:154]: @tofumatt now that I see this in context, we're now resetting 2/3 of the keys in the `NewComponent` state - the other being `error`. It seems that `error` should probably also be reset - essentially resetting all of the component state when the date range changes. What do you think? cc: @felixarntz @adamsilverstein

Comment (tofumatt) [reply assets/js/components/higherorder/withdata.js:154]: I was thinking that too, but I was just a bit wary to change more than needed. That said, resetting it shouldn't cause any issues. I'll try it out locally, but I think that's a good call and would prevent future issues similar to this one. 👍

Comment (tofumatt) [reply assets/js/components/higherorder/withdata.js]: Just tested this locally and it worked fine without causing issues, so I've added `error: false` to the `setState` call here. Ready for another look!

**Pull request merged**